### PR TITLE
split libs review rotation in two

### PR DESCRIPTION
**Pull request by yaahc**
The libs team discussed how we could improve review bandwidth for libs vs libs-api PRs and came to the conclusion that the best first step would be to separate the review rotations.

The goal of this split is to allow team members to focus on subsets of library work that they're most interested in and to allow independent growth of the review rotation focused on non-API affecting changes (aka libs PRs vs libs-api PRs). The libs APIs are generally less common, easier to review, often more time sensitive, and a much better entry point for new contributors interested in participating in the libs team. These properties make this an ideal starting point for new libs reviewers and contributors. By splitting the review rotations and focusing on increasing review bandwidth for non-API changes we can make this contribution experience more pleasant which will encourage repeat contributors and increase how much important bug fix / perf improvement / doc improvement style work we get done as a team.

### CODE REVIEW

**Review by dtolnay — COMMENTED** [review]
This would end up assigning all library PRs to `libs` due to this dirs mapping:

https://github.com/rust-lang/highfive/blob/9c691d552762ac1e643adae45bb06a5fa6141883/highfive/configs/rust-lang/rust.json#L103-L108

Maybe highfive could comment on library PRs instructing that if the PR contains a new library API or contains a stabilization of a library feature that has not already completed FCP in its tracking issue, then please comment with `r? rust-lang/libs-api @rustbot label +T-libs-api`

**Comment by yaahc** [comment]
> This would end up assigning all library PRs to `libs` due to this dirs mapping:

That was my intent, I figured as a starting point we could rely on the libs, team to reassign to libs-api when appropriate.

> Maybe highfive could comment on library PRs instructing that if the PR contains a new library API or contains a stabilization of a library feature that has not already completed FCP in its tracking issue, then please comment with `r? rust-lang/libs-api @rustbot label +T-libs-api`

That sounds like a great idea! I think there's a message field so I'll set that up.

File: highfive/configs/rust-lang/rust.json [hunk 1]
```diff
@@ -188,6 +193,10 @@
         "src/tools/miri": {
             "message": "Some changes occured to the Miri submodule",
             "reviewers": ["@rust-lang/miri"]
+        },
+        "library": {
+            "message": "Thank you for submitting a new PR for the library teams! If this PR contains a stabilization of a library feature that has not already completed FCP in its tracking issue, introduces new or changes existing unstable library APIs, or changes our public documentation in ways that create new stability guarantees for specific behavior then please comment with `r? rust-lang/libs-api @rustbot label +T-libs-api` to request review from a libs-api reviwer.",
+            "reviewers": []
```
Comment (yaahc) [inline highfive/configs/rust-lang/rust.json:199]: I double checked impl for highfive to make sure it would correctly format this message even if there are no reviewers. Seems like it's all fine and doesn't do much checking on the content of the `mentions` body until it gets to here:

* https://github.com/rust-lang/highfive/blob/master/highfive/newpr.py#L375-L380
* https://github.com/rust-lang/highfive/blob/master/highfive/newpr.py#L136

The one problem I did find is that it will print the `cc` unconditionally, so if we don't put anything in this mentions field won't look as nice.

```python
>>> "%scc %s" % ("hey thanks for the PR!\n\n", ','.join([]))
'hey thanks for the PR!\n\ncc '
```

I can change the impl to branch on if the reviewers list is empty and to only print the message if so, but I don't want to jump right to that since I feel like that might be an abuse of the "mentions" field and I don't want to turn this into a bunch of spaghetti.

**Comment by yaahc** [comment]
I wonder if this will work...

r? @Mark-Simulacrum

File: highfive/configs/rust-lang/rust.json (hunk 1 above)
Comment (Mark-Simulacrum) [reply highfive/configs/rust-lang/rust.json:199]: I think it makes sense to slightly adjust the impl as you suggest, and I would also add an explicit "no worries if you're not sure - the reviewer will take a look and make a decision to forward on if necessary"

Comment (Mark-Simulacrum) [reply highfive/configs/rust-lang/rust.json:199]: I agree that this is a minor "abuse" but it'll get cleanly solved once we migrate to triagebot doing this, for now it's ok.

File: highfive/configs/rust-lang/rust.json [hunk 2]
```diff
@@ -188,6 +193,10 @@
         "src/tools/miri": {
             "message": "Some changes occured to the Miri submodule",
             "reviewers": ["@rust-lang/miri"]
+        },
+        "library": {
+            "message": "Thank you for submitting a new PR for the library teams! If this PR contains a stabilization of a library feature that has not already completed FCP in its tracking issue, introduces new or changes existing unstable library APIs, or changes our public documentation in ways that create new stability guarantees then please comment with `r? rust-lang/libs-api @rustbot label +T-libs-api` to request review from a libs-api team reviewer. If you're unsure where your change falls no worries, just leave it as is and the reviewer will take a look and make a decision to forward on if necessary",
```
Comment (Mark-Simulacrum) [inline highfive/configs/rust-lang/rust.json:198]: I'm not sure I'm seeing code -- maybe we should add a test -- that the `r?` here isn't going to be picked up as a command. Triagebot has handling to ignore code blocks like this, but I don't think highfive does, so that won't be enough.

Comment (yaahc) [reply highfive/configs/rust-lang/rust.json:198]: > I'm not sure I'm seeing code -- maybe we should add a test -- that the `r?` here isn't going to be picked up as a command. Triagebot has handling to ignore code blocks like this, but I don't think highfive does, so that won't be enough.
> 
> An easy-ish fix might be to ignore things the bot itself posts, not sure if we make use of that anywhere.

Sounds good, I'll take a look and see if I can't add that.

Comment (yaahc) [reply highfive/configs/rust-lang/rust.json:198]: @Mark-Simulacrum looks like we're already good to go

https://github.com/rust-lang/highfive/blob/master/highfive/newpr.py#L438-L440

**Comment by Mark-Simulacrum** [comment]
Looks good with the conflict fixed.

File: highfive/configs/rust-lang/rust.json [hunk 3]
```diff
@@ -23,14 +23,19 @@
             "compiler-team-contributors"
         ],
         "libs": [
-            "@dtolnay",
```
Comment (Mark-Simulacrum) [inline highfive/configs/rust-lang/rust.json:26]: Is this change intentional? I don't see it in the discussion I think?

Comment (yaahc) [reply highfive/configs/rust-lang/rust.json:26]: Yea, the split was david's idea for helping improve the libs-api teams review bandwidth by making it so libs-api focused reviewers, using himself as an example, don't have to triage and review non API changes and can focus more time on API changing PRs.
